sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/implementation/models/ChatRetentionPolicy.java

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+
+package com.azure.communication.chat.implementation.models;
+
+import com.azure.core.annotation.Immutable;
+import com.azure.json.JsonReader;
+import com.azure.json.JsonSerializable;
+import com.azure.json.JsonToken;
+import com.azure.json.JsonWriter;
+import java.io.IOException;
+
+/**
+ * Data retention policy for auto deletion.
+ */
+@Immutable
+public class ChatRetentionPolicy implements JsonSerializable<ChatRetentionPolicy> {
+    /*
+     * Retention Policy Type
+     */
+    private RetentionPolicyKind kind = RetentionPolicyKind.fromString("ChatRetentionPolicy");
```

Comment:
If this is code fully generated by AutoRest disgregard my comment:

This class seems like a parent to extend from, but is there any case where the service could return a vanilla "ChatRetentionPolicy" that's not one of its children? Otherwise, couldn't this be an abstract class?

Reply:
I agree. Let me try I can change the autorest.md file to make it abstract class. If this cannot be made. I think we should keep it as it is. Otherwise, every time, autorest generation would undo the change.

Reply:
Not luck in my attempt to auto generate the class as abstract. I posted a question for this: https://teams.microsoft.com/l/message/19:e7dbf9972a574495a05cf2b9f0b3efbc@thread.skype/1748646079256?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1748646079256&teamName=Azure%20SDK&channelName=AutoRest%20-%20Partners&createdTime=1748646079256.

If it is too difficult to modify the md file to generate the policy file as abstract class. We might need to keep this as public class.

Reply:
They replied generating abstract class is not a supported feature by auto-rest. I would keep this class as implemented. Otherwise, the next time auto-rest would change it to implemented class.